prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt

```diff
@@ -0,0 +1,135 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7.utils
+
+import gov.cdc.prime.router.fhirengine.translation.hl7.HL7ConversionException
+import org.apache.commons.lang3.math.NumberUtils
+import org.apache.commons.text.StringSubstitutor
+import org.apache.commons.text.lookup.StringLookup
+import org.apache.logging.log4j.kotlin.Logging
+import org.hl7.fhir.exceptions.PathEngineException
+import org.hl7.fhir.r4.model.Base
+import org.hl7.fhir.r4.model.IntegerType
+import org.hl7.fhir.r4.model.StringType
+import org.hl7.fhir.r4.model.TypeDetails
+import org.hl7.fhir.r4.model.ValueSet
+import org.hl7.fhir.r4.utils.FHIRPathEngine
+
+/**
+ * Context used for resolving [constants].
+ */
+data class CustomContext(val constants: MutableMap<String, String> = mutableMapOf()) {
+    companion object {
+        /**
+         * Add [constants] to a context.
```

Comment:
Please update the note to indicate that this will simply return the existing context passed in if the element's constants are null. That will be helpful showing up in intellisense for people less familiar with the code.

Reply:
Makes sense. Added